hercules-ci-agent/hercules-ci-agent/Hercules/Agent/Cachix/Info.hs

```diff
@@ -0,0 +1,32 @@
+module Hercules.Agent.Cachix.Info where
+
+import           Protolude
+import qualified Cachix.Client.Config          as Cachix.Config
+import qualified Cachix.Client.Env             as Cachix.Env
+import           Hercules.Agent.Log
+import           Hercules.Agent.Cachix.Env
+import qualified Katip.Core                    as K
+
+getAvailablePushCaches :: (MonadReader r m, HasEnv r) => m [Text]
+getAvailablePushCaches = do
+  env <- asks (cachixEnv . getEnv)
+  let
+    pushableName Cachix.Config.BinaryCacheConfig { secretKey = _required, name = name }
+      = [name]
+
+  pure $ do -- []
+    config <- toList $ Cachix.Env.config env
+    cache <- toList $ Cachix.Config.binaryCaches config
+    pushableName cache
+
+activePushCaches :: (MonadReader r m, HasEnv r) => m [Text]
+activePushCaches = asks (pushCaches . getEnv)
+
+validate :: (MonadReader r m, HasEnv r, Katip m) => m ()
```

Comment:
It's documented and public interface to use `CACHIX_SIGNING_KEY=... cachix push foo` without having a configuration file. The reasoning is that the key might have been generated on another machine.

`validate` will mislead the user that cache won't work, although it will.

I don't think we have a choice but to support `CACHIX_SIGNING_KEY=` as there's no way to write cachix configuration file given the signing key atm.